recipes/liquid-dsp/all/conanfile.py

```diff
@@ -0,0 +1,178 @@
+            for patch in self.conan_data["patches"][self.version]:
+                tools.patch(**patch)
+
+    def _gen_link_library(self):
+        if self.settings.compiler != "Visual Studio" or (not self.options.shared):
+            return
+        self.run("cmd /c generate_link_library.bat")
+        with tools.chdir(self._source_subfolder):
+            self.run(
+                f"""{os.getenv("AR")} /def:libliquid.def /out:libliquid.lib """
+                f"""/machine:{"X86" if self.settings.arch=="x86" else "X64"}""",
+                win_bash=tools.os_info.is_windows,
+            )
+
+    def _rename_libraries(self):
+        with tools.chdir(self._source_subfolder):
+            if self.settings.os == "Windows" and self.options.shared:
+                os.rename("libliquid.so", "libliquid.dll")
+            elif self.settings.os == "Windows" and not self.options.shared:
+                os.rename("libliquid.a", "libliquid.lib")
+            elif self.settings.os == "Macos" and not self.options.shared:
+                os.rename("libliquid.ar", "libliquid.a")
+
+    @contextmanager
+    def _build_context(self):
+        if self.settings.compiler == "Visual Studio":
+            env = {
+                "CC": "gcc",
+                "CXX": "g++",
+                "LD": "ld",
+                "AR": "ar",
+            }
+            with tools.environment_append(env):
+                yield
+        else:
+            yield
+
+    @contextmanager
+    def _msvc_context(self):
+        if self.settings.compiler == "Visual Studio":
+            with tools.vcvars(self.settings):
+                env = {
+                    "CC": "cl -nologo",
+                    "CXX": "cl -nologo",
+                    "AR": "lib",
+                    "LD": "link",
+                }
+                with tools.environment_append(env):
+                    yield
+        else:
+            yield
+
+    def build(self):
+        self._patch_sources()
+        ncpus = os.cpu_count()
+        ncpus = ncpus if ncpus else 1
+        configure_args = []
+        if self.settings.build_type == "Debug":
+            configure_args.append("--enable-debug-messages")
+        if self.options.simdoverride:
+            configure_args.append("--enable-simdoverride")
+        if self.settings.compiler == "Visual Studio":
+            configure_args.append("CFLAGS='-static-libgcc'")
+        configure_args_str = " ".join(configure_args)
+        with self._build_context():
+            with tools.chdir(self._source_subfolder):
+                self.run("./bootstrap.sh", win_bash=tools.os_info.is_windows)
+                self.run(
+                    f"""./configure {configure_args_str}""",
+                    win_bash=tools.os_info.is_windows,
+                )
+                self.run(
+                    f"""make {self._target_name} -j{ncpus}""",
+                    win_bash=tools.os_info.is_windows,
+                )
```

Comment:
```suggestion
                self.run(
                    f"""./configure {configure_args_str}""",
                    win_bash=tools.os_info.is_windows,
                )
                self.run(
                    f"""make {self._target_name} -j{ncpus}""",
                    win_bash=tools.os_info.is_windows,
                )
```

Use AutoToolsBuildEnvironment instead. Take a look on https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md#how-to-provide-a-good-recipe for real examples using autotools.

Reply:
Yes, I knew about it, but I am not using it on purpose. Notice the recipe is a bit "hacky". It actually builds with mingw, even if the selected compiler is MSVC due to the lack of C99 complex.h support on MSVC. With the AutoToolsBuildEnvironment I couldn't get it to change the build environment as I wanted it to. You can see [how bad it was](https://github.com/conan-io/conan-center-index/blob/baa6190b1a7020b23343faa2d4f6ca9ea2c6425d/recipes/liquid-dsp/all/conanfile.py)...

Reply:
I tried things like:
```python
       have_to_rollback = False
        if self.settings.compiler == "Visual Studio":
            # Save current information
            old_compiler = str(self.settings.compiler)
            old_compiler_runtime = str(self.settings.compiler.runtime)
            old_compiler_toolset = str(self.settings.compiler.toolset)

            self.settings.compiler = "gcc"
            self.settings.compiler.version = "8.1"
            self.settings.compiler.threads = "win32"

            have_to_rollback = True
```

I tried to copy the ConanFile (self) object and change that one, but ConanFile objects cannot be deep-copied because of the use of StringIO. So yeah, I found myself out of options and decided to do it manually instead.

Reply:
Wouldn't it be simpler to use autotoolsbuildhelper, and declare MINGW to be compatible with MSVC using https://docs.conan.io/en/latest/creating_packages/define_abi_compatibility.html#compatible-packages ?

Reply:
The MSVC package needs aditional steps (generating the link-library for example). Not sure if that's possible with that feature. Thanks for bringing it up to my attention. I'll explore that. 

If it's okay for you I think I prefer to have the recipe accepted as-is and think about that later...